client/src/components/History/ContentItem/HistoryContentItem.vue

```diff
@@ -0,0 +1,106 @@
+<!-- For each element in the history, this selects whether it will be displayed as a dataset or a dataset collection -->
+
+<template>
+    <component
+        :is="contentItemComponent"
+        class="content-item p-1"
+        v-on="$listeners"
+        v-bind="$attrs"
+        :index="index"
+        :tabindex="index"
+        :item="localItem"
+        @update:item="onUpdate"
+        @delete="onDelete"
+        @undelete="onUndelete"
```

Comment:
This seems to not be defined now?
<img width="1213" alt="Screenshot 2021-04-08 at 12 07 05" src="https://user-images.githubusercontent.com/6804901/114009117-28e1c100-9863-11eb-9988-809bdc7cafa8.png">

If I remove both then I'm getting
<img width="1161" alt="Screenshot 2021-04-08 at 12 31 21" src="https://user-images.githubusercontent.com/6804901/114012229-5b40ed80-9866-11eb-882a-413c76caea7c.png">

I guess this still needs some tweaking ?
Overall this looks like a nice improvement.

Reply:
That's weird, I'm looking at my code right now and onUndelete is clearly defined in that component. I'll look at it closer after diner.